contracts/RMRK/extension/tokenProperties/RMRKTokenPropertiesRepository.sol

```diff
@@ -0,0 +1,286 @@
+     * @param tokenId The token ID
+     * @param key The property key
+     * @param value The property value
+     */
+    function _setStringProperty(
+        address collection,
+        uint256 tokenId,
+        string memory key,
+        string memory value
+    )
+        internal
+        onlyAuthorizedCaller(
+            collection,
+            ParameterType.STRING,
+            _getIdForKey(collection, key),
+            tokenId
+        )
+    {
+        _stringValueIds[collection][tokenId][
+            _getIdForKey(collection, key)
+        ] = _getStringIdForValue(collection, value);
+        emit StringPropertyUpdated(collection, tokenId, key, value);
+    }
+
+    /**
+     * @notice Used to set a boolean property.
+     * @param collection Address of the collection receiving the property
+     * @param tokenId The token ID
+     * @param key The property key
+     * @param value The property value
+     */
+    function _setBoolProperty(
+        address collection,
+        uint256 tokenId,
+        string memory key,
+        bool value
+    )
+        internal
+        onlyAuthorizedCaller(
+            collection,
+            ParameterType.BOOL,
+            _getIdForKey(collection, key),
+            tokenId
+        )
+    {
+        _boolValues[collection][tokenId][_getIdForKey(collection, key)] = value;
+        emit BoolPropertyUpdated(collection, tokenId, key, value);
+    }
+
+    /**
+     * @notice Used to set an bytes property.
+     * @param collection Address of the collection receiving the property
+     * @param tokenId The token ID
+     * @param key The property key
+     * @param value The property value
+     */
+    function _setBytesProperty(
+        address collection,
+        uint256 tokenId,
+        string memory key,
+        bytes memory value
+    )
+        internal
+        onlyAuthorizedCaller(
+            collection,
+            ParameterType.BYTES,
+            _getIdForKey(collection, key),
+            tokenId
+        )
+    {
+        _bytesValues[collection][tokenId][
+            _getIdForKey(collection, key)
+        ] = value;
+        emit BytesPropertyUpdated(collection, tokenId, key, value);
+    }
+
+    /**
+     * @notice Used to set an address property.
+     * @param collection Address of the collection receiving the property
+     * @param tokenId The token ID
+     * @param key The property key
+     * @param value The property value
+     */
+    function _setAddressProperty(
+        address collection,
+        uint256 tokenId,
+        string memory key,
+        address value
+    )
+        internal
+        onlyAuthorizedCaller(
+            collection,
+            ParameterType.ADDRESS,
+            _getIdForKey(collection, key),
+            tokenId
+        )
+    {
+        _addressValues[collection][tokenId][
+            _getIdForKey(collection, key)
+        ] = value;
+        emit AddressPropertyUpdated(collection, tokenId, key, value);
+    }
```

Comment:
I think all this setters are prefect with one caveat. Since this is not an extensions, these should already be the external versions.

Reply:
🦾